src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -482,4 +492,78 @@ export class StudyViewPageStore {
         },
         default: {}
     });
+
+    readonly mutationCounts = remoteData<MutationCount[]>({
+        await:()=>[
+            this.samples
+        ],
+        invoke:async()=>{
+            const studyToSamples = _.groupBy(this.samples.result!, s=>s.studyId);
+            return _.flatten(await Promise.all(
+                this.studyIds.map(studyId=>{
```

Comment:
instead of looping `studyIds` may be better to loop `mutationProfiles`

Reply:
it seems equivalent to me, whats your thought process?

Reply:
something like
```
return _.flatten(await Promise.all(
    _.map(this.mutationProfiles, (profile) => {
        const samples = studyToSamples[profile.studyId];
        return defaultClient.fetchMutationCountsInMolecularProfileUsingPOST({
            molecularProfileId: profile.molecularProfileId,
            sampleIds: samples.map(s=>s.sampleId)
        });
    })
));
```

Reply:
no I understand but why do you prefer that?

Reply:
I prefer this way instead of using null and filtering them

Reply:
oh I see. that makes sense to me, i will change